### PR TITLE
Fixed animation

### LINKED ISSUE #1855: Flickering text when expanding info from 'help' menu.
**Describe the bug**

When expanding "What does kiwix do?" and "Where is the content?" the text flickers by going up and down fast. (See vid. below)

**Expected behavior**

The expanded text/view should not flicker.

**Steps to reproduce the behavior:**
1. Open kiwix app.
2. Click three dots in upper right corner.
3. Click the help option.
4. Expand "What does kiwix do?" or "Where is the content?" a couple of times. The expanded views should flicker. 

**Screenshots**

[Video with higher framerate, as the issue is not visible in 10fps](https://photos.app.goo.gl/2n2CafAUJoNT4iQ77)

**Environment**
- Version of Kiwix Android : 3.2.1 Build: 6230201
- Device : OnePlus 5
- OS version : 9.0.10 OxygenOS

### PR DESCRIPTION
**Pull request by sonusourav**
Fixes #1855 

## Changes
Added `description.setVisibility(View.GONE);` and 
`description.setVisibility(View.VISIBLE); `
It results into the same thing as expected.

### CODE REVIEW

**Comment by abdulwd** [comment]
Is this just disabling the animation?

**Comment by sonusourav** [comment]
> 
> 
> Is this just disabling the animation?

Yeah, I observed that doing this has the same effect as with animation.

File: core/src/main/java/org/kiwix/kiwixmobile/core/help/HelpAdapter.java [hunk 1]
```diff
@@ -33,9 +33,6 @@
 import org.kiwix.kiwixmobile.core.R;
 import org.kiwix.kiwixmobile.core.R2;
 
-import static org.kiwix.kiwixmobile.core.utils.AnimationUtils.collapse;
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/help/HelpAdapter.java]: can these methods be deleted now?

Comment (sonusourav) [reply core/src/main/java/org/kiwix/kiwixmobile/core/help/HelpAdapter.java]: Yeah removed.

**Comment by abdulwd** [comment]
I think we should fix the animation instead of removing it altogether.

**Comment by sonusourav** [comment]
@abdulwd Could you please help me know how should the animation look like. Because, without animation also, it looks nearly the same.

**Comment by abdulwd** [comment]
https://medium.com/better-programming/recyclerview-expanded-1c1be424282c

**Comment by sonusourav** [comment]
@abdulwd Please review.

**Comment by macgills** [comment]
@abdulwd We should turn on `animatLayoutChanges` and not write basic animations ourselves when the framework will just do it for us

**Comment by sonusourav** [comment]
> 
> 
> @abdulwd We should turn on `animatLayoutChanges` and not write basic animations ourselves when the framework will just do it for us

@macgills  I tried `animateLayoutChanges` but the flickering was still there. I feel that the animation is infact not required here.

**Comment by macgills** [comment]
@sonusourav That seems unlikely to me. We use `animateLayoutChanges` for long presses on your local library to animate in the checkbox to select for share/delete, it is buttery smooth.

Can you post some gifs of the results of the various changes?

**Comment by sonusourav** [comment]
> 
> 
> @sonusourav That seems unlikely to me. We use `animateLayoutChanges` for long presses on your local library to animate in the checkbox to select for share/delete, it is buttery smooth.
> 
> Can you post some gifs of the results of the various changes?

Here is it
Case 1: [When `animateLayoutChanges=true` and animation is added](https://imgur.com/DdvQYRS)

Case 2: [When `animateLayoutChanges=true` and no animation is added](https://imgur.com/4nfSqlt)

Case 3: [Without any animation and `animateLayoutChanges=false`](https://imgur.com/csdj6t8)

**Comment by macgills** [comment]
I don't like any of those options.

Stop this being a Recyclerview. You can make these things Custom Views and set them up to take attributes so we can statically declare them in xml. 

The app makes liberal use of `animateLayoutChanges` and that is all this should be. If these were all just in a normal layout it should be no problem.

**Comment by sonusourav** [comment]
@macgills So, what should I implement finally?
Just setting `animateLayoutChanges =true` without adding any animation is giving [case 2 result](https://imgur.com/4nfSqlt). What should I proceed with?

**Comment by macgills** [comment]
@sonusourav 

>Stop this being a Recyclerview. You can make these things Custom Views and set them up to take attributes so we can statically declare them in xml.

Use animateLayoutChanges. If the above idea fails then I guess it is back to fixing the animations

**Comment by sonusourav** [comment]
> 
> 
> @sonusourav
> 
> > Stop this being a Recyclerview. You can make these things Custom Views and set them up to take attributes so we can statically declare them in xml.
> 
> Use animateLayoutChanges. If the above idea fails then I guess it is back to fixing the animations

With just animate layout, the animation does not look good. Here is how it is looking.
[Link](https://imgur.com/8tsABvi)
So, I have just corrected the animation. Let me know your views.

**Comment by macgills** [comment]
@sonusourav It will look fine if the content was not in a recyclerview

>Stop this being a Recyclerview. You can make these things Custom Views and set them up to take attributes so we can statically declare them in xml.

Do you have questions about the above?

**Comment by sonusourav** [comment]
> 
> 
> @sonusourav It will look fine if the content was not in a recyclerview
> 
> > Stop this being a Recyclerview. You can make these things Custom Views and set them up to take attributes so we can statically declare them in xml.
> 
> Do you have questions about the above?

So, should I remove recycler view and use something else like textview for this?

**Comment by macgills** [comment]
[Custom Views](https://developer.android.com/guide/topics/ui/custom-components), see `TagsView` if you need a little inspiration, I want it to look identical to how it is now

**Comment by sonusourav** [comment]
> 
> 
> [Custom Views](https://developer.android.com/guide/topics/ui/custom-components), see `TagsView` if you need a little inspiration, I want it to look identical to how it is now

Hey @macgills , I was thinking its fine to use recycler view here because if the number of help items increases, its easier to append the new item in recycler view while in other views, we have to add many lines of code to achieve the same. And in the end, the UI is anyway not going to change, as we want. But still if you feel we should change this, let me know.

**Comment by macgills** [comment]
Yeah I was wrong.The setting of a view gone does look prettty fucky from some quick experiments I ran even with enabling the changing transition type. You could probably fix this by using animateLayoutChanges and [altering the animation it uses](https://stackoverflow.com/questions/35174753/how-to-change-animations-used-by-animatelayoutchanges-mechanism).

**Comment by sonusourav** [comment]
> 
> 
> Yeah I was wrong.The setting of a view gone does look prettty fucky from some quick experiments I ran even with enabling the changing transition type. You could probably fix this by using animateLayoutChanges and [altering the animation it uses](https://stackoverflow.com/questions/35174753/how-to-change-animations-used-by-animatelayoutchanges-mechanism).

Hey @macgills , the code in above link does not work. And I feel that the current expand and collapse animation (after making the PR) works fine.

**Pull request merged**